Recon/src/Recon/Skims/RecoilFiducialityProcessor.cxx

```diff
@@ -0,0 +1,148 @@
+  auto particleMap{
+      event.getMap<int, ldmx::SimParticle>("SimParticles", input_pass_name_)};
+
+  // Search for the recoil electron
+  auto [recoil_track_id, recoil_electron] = Analysis::getRecoil(particleMap);
+
+  // Get the collection of simulated Ecal hits from the event.
+  const std::vector<ldmx::SimCalorimeterHit> ecal_sim_hits =
+      event.getCollection<ldmx::SimCalorimeterHit>(ecal_collection_,
+                                                   input_pass_name_);
+
+  // Get the collection of simulated Ecal hits from the event.
+  const std::vector<ldmx::SimCalorimeterHit> hcal_sim_hits =
+      event.getCollection<ldmx::SimCalorimeterHit>(hcal_collection_,
+                                                   input_pass_name_);
+
+  // Get the collection of simulated tracker hits from the event.
+  const std::vector<ldmx::SimTrackerHit> recoil_sim_hits =
+      event.getCollection<ldmx::SimTrackerHit>(recoil_collection_,
+                                               input_pass_name_);
+
+  // Loop through the Ecal hits and check if the recoil electron is
+  // associated with any of them.
+  bool has_ecal_hit = false;
+  int ecal_hit_id = -1;
+  for (const ldmx::SimCalorimeterHit &sim_hit : ecal_sim_hits) {
+    for (int iContrib = 0; iContrib < sim_hit.getNumberOfContribs();
+         ++iContrib) {
+      ldmx::SimCalorimeterHit::Contrib contrib = sim_hit.getContrib(iContrib);
+
+      if (contrib.trackID == recoil_track_id) {
+        has_ecal_hit = true;
+        ecal_hit_id = sim_hit.getID();
+      }
+    }
+  }
+
+  // Loop through the Hcal hits and check if the recoil electron is
+  // associated with any of them.
+  bool has_hcal_hit = false;
+  int hcal_hit_id = -1;
+  for (const ldmx::SimCalorimeterHit &sim_hit : hcal_sim_hits) {
+    for (int iContrib = 0; iContrib < sim_hit.getNumberOfContribs();
+         ++iContrib) {
+      ldmx::SimCalorimeterHit::Contrib contrib = sim_hit.getContrib(iContrib);
+
+      if (contrib.trackID == recoil_track_id) {
+        has_hcal_hit = true;
+        hcal_hit_id = sim_hit.getID();
+      }
+    }
+  }
+
+  // Loop through the recoil tracker hits and count how many
+  // the recoil electron is associated with
+  std::set<int> layers_hit;
+  for (const ldmx::SimTrackerHit &sim_hit : recoil_sim_hits) {
+    if (sim_hit.getTrackID() == recoil_track_id) {
+      // int sensorID = tracking::sim::utils::getSensorID(sim_hit);
+      if ((sim_hit.getTime() < 0.8) && (sim_hit.getMomentum()[2] > 0)) {
+        layers_hit.insert(sim_hit.getLayerID());
+      }
+    }
+  }
+  bool has_min_tracker_hits = false;
+  if (layers_hit.size() >= min_tracker_hits_) {
+    has_min_tracker_hits = true;
+  }
+
+  // Checking if the recoil electron energy is > min energy
+  bool has_min_energy = false;
+  if (recoil_electron->getEnergy() >= min_p_mag_) {
+    has_min_energy = true;
+  }
+
+  // Configure outputs
+  bool is_fiducial = has_min_energy & has_min_tracker_hits & has_ecal_hit;
```

Comment:
I wonder if this was supposed to have been
```
bool is_fiducial = has_min_energy && has_min_tracker_hits && has_ecal_hit;
```
since it's not a bitwise operation? @EBerzin

Reply:
Fix in https://github.com/LDMX-Software/ldmx-sw/pull/1641